docs/benchmark.md

```diff
@@ -0,0 +1,108 @@
+# Benchmark baseline results
+
+This document contains baseline benchmark results for Observatorium API Gateway under synthetic load.
+
+Tested on:
+
+```txt
+(15-inch, 2019)
+2,6 GHz 6-Core Intel Core i7
+16 GB 2400 MHz DDR4
+```
+
+Generated using:
+
+[embedmd]:# (../tmp/load_help.txt)
+```txt
+load.sh [-h] [-r n] [-c n] [-m n] [-q n] [-o csv|gnuplot] -- program to test synthetic load on observatorium gateway and report results.
+
+where:
+    -h  show this help text
+    -r  set number of seconds to run (default: 300)
+    -c  set number of cluster to simulate (default: 5000)
+    -m  set number of machines per cluster to simulate (default: 2)
+    -q  set number of concurrent queries to execute (default: 10)
+    -o  set the output format (default: csv. options: csv, gnuplot)
+```
+
+With parameters:
+
+```console
+$ ./test/load.sh -r 300 -c 1000 -m 3 -q 10 -o gnuplot
+```
+
+> It runs tets for 5 minutes, simulating 3000 machines sending metrics and 10 consumers querying for their data every second.
+> Observatorim API GW runs in-front of 1 Thanos Receive and 1 Thanos Querier.
+
+## Results
+
+Most relevant results are the ones on resource consumption.
+CPU usage is pretty much stable.
+Memory usage correlates with number of goroutine, which correlates number of open connections.
+Memory usage increases and request latencies increase as the backend services' load increase, which is expected.
+
+### Resource consumption
+
+#### CPU Usage
+
+> `rate(process_cpu_seconds_total{job="observatorium"}[1m]) * 100`
+
+![./loadtests/cpu.png](./loadtests/cpu.png)
+
+#### Memory Usage
+
+> `process_resident_memory_bytes{job="observatorium"}'`
```

Comment:
Not a big deal but wonder where those 200MB came from

Reply:
I don't have in-depth knowledge about reverse proxy in go, however, this is an interesting point to check out. I'm going to investigate it.

Reply:
I have checked it again, it's not 200MB rather 20MB.

Reply:
A snapshot of the heap: 
![profile004](https://user-images.githubusercontent.com/536449/78160741-a8fd6b80-7444-11ea-972a-5c66a5297d18.png)